### DISCUSSION TITLE
Required Data collection

### DISCUSSION THREAD

**Discussion by nemishzalavadiya-pirimid** [ONDC-Official/ONDC-FIS-Specifications#23] [Q&A] [answered]
Draft [link](https://github.com/ONDC-Official/ONDC-FIS-Specifications/tree/draft-2.x)
Swagger [link](https://ondc-official.github.io/ONDC-FIS-Specifications/)

How form will work on buyer app shared in response by Seller app

> "form": {
>               "mime_type": "text/html",
>               "url": "https://6vs8xnx5i7.icicibank.co.in/loan-agreement/xinput/formid/a23f2fdfbbb8ac402bfd54f",
>               "resubmit": false,
>               "multiple_sumbissions": false
>             }

here this form -> url should be a UI page which will be rendered on buyer app to capture data, please confirm?

**Comment by Antriksh-Parmar** [comment]
@92shreyansh

**Comment by 92shreyansh** [comment]
These would be text/html format as html forms or full fleged UI.
For html forms it would be a choice of the BAP if they want to parse and render in their own format UI or render it as is.
For complete UI page or html it would again be a choice on BAP if they want to embeed within the UI or redirect to browser etc.

These can be identified via mime_type application/html for complete UI page and text/html for html forms will update the examples accordingly.

**Reply by nemishzalavadiya-pirimid** [reply]
So just to clarify, 
if it's fully **fledged UI**, then data submition will be handled by UI itself **Buyer app** will not have any control, so in such case how will Buyer app know that data is been submitted or not?

will buyer app directly wait for **on_init** response without **init** request?

cases
1. if embeded within UI -> How buyer app will know that data is been submitted, as there is no status check API
2. if redirect to browser how/where to redirect back on sucessful submission?

Can you please share for clarity?

**Reply by Antriksh-Parmar** [reply]
@92shreyansh

**Reply by 92shreyansh** [reply, marked as answer]
Please refer the form status for application/html for how status APIs will be used to resolve this

**Reply by nemishzalavadiya-pirimid** [reply]
FYI for ONDC team,

You are assuming that BAP app will use iframe to load `appication/html` urls, but let' say for emandate if BPP uses Razorpay then iframe randoring is not allowed, Due to RBI safty rules.

> How to do Emandate in that case?


Or Let's say while doing emandate if it's redirection approch from BPA side then also need `redirectionBack` URL after emandate completes.

Please consider this as well.

**Reply by 92shreyansh** [reply]
This has been incorporated in Mutual Funds and Credit 2.3.0 by having a standard /callback call to be done form Lenders UI to the base subscriber_url
